lib/msal-common/src/request/CommonDeviceCodeRequest.ts

```diff
@@ -16,8 +16,9 @@ import { BaseAuthRequest } from "./BaseAuthRequest";
  * - resourceRequestMethod      - HTTP Request type used to request data from the resource (i.e. "GET", "POST", etc.).  Used for proof-of-possession flows.
  * - resourceRequestUri         - URI that token will be used for. Used for proof-of-possession flows.
  * - timeout                    - Timeout period in seconds which the user explicitly configures for the polling of the device code endpoint. At the end of this period; assuming the device code has not expired yet; the device code polling is stopped and the request cancelled. The device code expiration window will always take precedence over this set period.
+ * - tokenQueryParameters       - String to string map of custom query parameters added to the /token call
```

Comment:
Since this does not use the /token endpoint, consider omitting this here and exposing a more appropriately named parameter instead. Perhaps `extraQueryParameters` would work?

Reply:
Unless we keep it named tokenQueryParameters, I think this would cause too many complications. We'd have to add extraQueryParameters to BaseAuthRequest just so that it can be used in BaseClient's createTokenQueryParameters function. Then we'd have to omit extraQueryParameters in all the other requests. We'd also have to add support/documentation that the tokenQueryParameters should be passed in extraQueryParameters in the app's config file. Leaving it named tokenQueryParameters would eliminate these problems, I think. Thoughts?

Reply:
> We'd also have to add support/documentation that the tokenQueryParameters should be passed in extraQueryParameters in the app's config file.

I'm not sure I understand this concern. Can you please elaborate.

In any event the query string is constructed [in DeviceCodeClient](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/58e57222cfde9ccd20dd839785f09bb4a07c3a06/lib/msal-common/src/client/DeviceCodeClient.ts#L127) I don't think you need to do anything to `BaseAuthRequest`

Reply:
I understand now, thanks